uninstall.sh

```diff
@@ -46,18 +50,115 @@ printf "\n"
 oc cluster-info | head -n 1 | awk '{print $NF}'
 printf "\n"
 
-./clean-clusters.sh "$args"
-
-kubectl delete -k multiclusterhub/
-echo "Sleeping for 200 seconds to allow resources to finalize ..."
-sleep 200
+#./clean-clusters.sh "$args"
 
-kubectl delete -k multicluster-hub-operator/
-./multicluster-hub-operator/uninstall.sh
-
-kubectl delete -k acm-operator/
-./acm-operator/uninstall.sh
+if [ -z ${OCM_NAMESPACE+x} ]; 
+then 
+    echo "OCM_NAMESPACE must be set"
+    exit 1
+else 
+    echo "Cleaning up resources in namespace: $OCM_NAMESPACE"
+fi
 
-kubectl delete -k community-subscriptions/
+oc delete mco --all -n $OCM_NAMESPACE
+
+COMPLETE=1
+if [[ " $@ " =~ " --watch " ]]; then
+    if [[ $DEFAULT_SNAPSHOT =~ v{0,1}2\.[1-9][0-9]*\.[0-9]+.* ]]; then
+        echo ""
+        echo "#####"
+        #mch_status=`oc get multiclusterhub --all-namespaces`
+        acc=0
+	while [[ $(oc get multiclusterobservability --all-namespaces) && $acc -le $POLL_DURATION_21X ]]; do
+	    curr_status=$(oc get multiclusterobservability --all-namespaces -o json | jq -r '.items[].status.phase') 2> /dev/null
+            echo "Waited $acc/$POLL_DURATION_21X seconds for MCO to be deleted.  Current Status: $curr_status"
+            if [[ "$DEBUG" == "true" ]]; then
+                echo "#####"
+                component_list=$(oc get multiclusterobservability --all-namespaces -o json | jq -r '.items[].status.components')
+                printf "%-30s\t%-10s\t%-30s\t%-30s\n" "COMPONENT" "STATUS" "TYPE" "REASON"
+                for status_item in $(echo $component_list | jq -r 'keys | .[]'); do
+                    component=$(echo $component_list | jq -r --arg ITEM_NAME "$status_item" '.[$ITEM_NAME]')
+                    component_status="$(echo $component | jq -r '.status')";
+                    type="$(echo $component | jq -r '.type')";
+                    reason="$(echo $component | jq -r '.reason')";
+                    message="$(echo $component | jq -r '.message')";
+                    printf "%-30s\t%-10s\t%-30s\t%-30s\n" "$status_item" "$component_status" "$type" "$reason"
+                done
+            fi
```

Comment:
this doesnt work for obs so lets remove it

Reply:
Does observability not have components we can watch be torn down?

Reply:
probably not check their schema